### PR TITLE
Add proxy_lib_is_from_UMF() to the proxy_lib API

### PR DESCRIPTION
**Pull request by ldorau**
### Description

Add `proxy_lib_is_from_UMF()` to the proxy_lib API.

### Checklist
- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly

### CODE REVIEW

**Comment by ldorau** [comment]
Not needed
@bratpiorka

File: src/proxy_lib/proxy_lib.h [hunk 1]
```diff
@@ -14,6 +14,7 @@ extern "C" {
 
 void proxy_lib_create_common(void);
 void proxy_lib_destroy_common(void);
+int proxy_lib_is_from_UMF(void *ptr);
```
Comment (vinser52) [inline src/proxy_lib/proxy_lib.h:17]: How will this API be used and why can the client not use the `umfPoolByPtr()` function from UMF?

Comment (ldorau) [reply src/proxy_lib/proxy_lib.h:17]: It is not needed, see: https://github.com/oneapi-src/unified-memory-framework/pull/681#issuecomment-2324149846

Comment (ldorau) [reply src/proxy_lib/proxy_lib.h:17]: This PR had been closed.

**Pull request closed without merging**